src/Engine/ProtoCore/FFI/CLRObjectMarshler.cs

```diff
@@ -494,24 +494,21 @@ private ICollection ToICollection(StackValue dsObject, ProtoCore.Runtime.Context
     /// <summary>
     /// Marshales string as array of chars
     /// </summary>
-    class StringMarshaler : CollectionMarshaler
+    class StringMarshaler : PrimitiveMarshler 
```

Comment:
Verified with Revit's ``Element.ByParameters``, I think it should go through marshaler?